sql/catalyst/src/main/scala/org/apache/spark/sql/types/StructType.scala

```diff
@@ -359,6 +361,18 @@ object StructType extends AbstractDataType {
   protected[sql] def fromAttributes(attributes: Seq[Attribute]): StructType =
     StructType(attributes.map(a => StructField(a.name, a.dataType, a.nullable, a.metadata)))
 
+  def removeMetadata(key: String, dt: DataType): DataType =
+    dt match {
+      case StructType(fields) =>
+        val newFields = fields.map { f =>
+          val mb = new MetadataBuilder()
```

Comment:
This workaround may be taken out in the future.

However, use of MetadataBuilder occurs in many other places:
http://pastebin.com/nVjNfrgp

I feel adding clear() to MetadataBuilder would help in current and future use cases.